build/helper/metadata_add_all.py

```diff
@@ -109,36 +109,57 @@ def _add_is_error_handling(f):
     return f
 
 
-def _add_buffer_info(parameter):
-    '''Adds buffer information to the parameter metadata iff 'size' is defined else assume not a buffer'''
-
-    # For simplicity, we are going to treat ViChar[], ViString, ViConstString, and ViRsrc the same: As ViChar
-    # and is_buffer True
-    t = parameter['type']
-    if (t.find('[ ]') > 0) or (t.find('[]') > 0):
-        assert 'is_buffer' not in parameter or parameter['is_buffer'] is True, 'Conflicting metadata - [] found but is_buffer already set to False.'
-        parameter['type'] = t.replace('[ ]', '').replace('[]', '')
-        parameter['original_type'] = t
-        parameter['is_buffer'] = True
+def _add_buffer_info(parameter, config):
```

Comment:
I think is_string should be set in a separate `_add_is_string` function.

Reply:
I can do that but then in `_add_buffer_info()` I have to filter out `ViChar[]` since string doesn't change `['type']`. The logic here handles that.